solutions/usaco-839.mdx

```diff
@@ -0,0 +1,57 @@
+---
+id: usaco-839
+source: USACO Gold 2018 Open
+title: Talent Show
+author: Arpan Banerjee
+---
+
+[Official Editorial](http://www.usaco.org/current/data/sol_talent_gold_open18.html)
+
+<Warning title="Common Misconception">
+Instead of using binary search as the official solution does, a knapsack can be done. An array of talents can be kept, where `talent[i]` is the minimum weight required to create a talent sum of i. To make sure each cow is used at most once and the order of the cows doesn't matter, the indices of `talent` are iterated though from greatest to least, and the `cows` are iterated though in the outer loop; the implementation should make this clear. This maximizes the ratio $t/w$ as desired. Then, the maximum ratio of $t_i / w_i$ can be easily found by going through `talent` and ignoring any $w_i$ that is less than $W$. 
```

Comment:
```suggestion
Rather than using binary search, we can also use knapsack DP. We can initialize an array called 'talent', where `talent[i]` is the minimum weight required to create a talent sum of i. We can iterate 'talent' from greatest to least, and iterate the "cows" through the outer loop. This maximizes the ratio $t/w$ as desired. The maximum ratio of $t_i / w_i$ can be easily found by iterating through `talent` and ignoring any $w_i$ that is less than $W$. 
```

Reply:
This is effectively just rewording what I wrote. It does not add anything; in fact, it detracts from the original meaning slightly, in my opinion.